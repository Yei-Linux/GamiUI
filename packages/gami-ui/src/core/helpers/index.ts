export * from './date-formatter'
export * from './validations.helper'
