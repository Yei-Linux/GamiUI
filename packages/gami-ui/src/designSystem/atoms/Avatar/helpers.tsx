import { TextModeType } from 'core/domain/types'
import { avatarTextModes } from './constants'
import { REGEX_RULES } from 'core/utils/constants'

export const validateMaxLetters = (
  firstLettersGot: string,
  maxSizeLetters: number
) => {
  const size = firstLettersGot.length

  if (size > maxSizeLetters) {
    return firstLettersGot.substring(0, maxSizeLetters)
  }

  return firstLettersGot
}

export const showOnlyFirstLettersOnText = (
  text: string,
  type: TextModeType,
  maxSizeLetters: number
): string => {
  const regexType = avatarTextModes[type]
  const firstLettersCaptured: string[] | null = text.match(
    REGEX_RULES[regexType]
  )

  const joinFirstLetters = firstLettersCaptured?.join('') || ''

  return validateMaxLetters(joinFirstLetters, maxSizeLetters)
}
