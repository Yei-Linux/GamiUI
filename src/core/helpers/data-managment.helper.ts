export const mapEnum = (enumerable: any): any[] => {
  let enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  );
  return enumMembers;
};

