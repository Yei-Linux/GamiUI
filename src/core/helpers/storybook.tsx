import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};