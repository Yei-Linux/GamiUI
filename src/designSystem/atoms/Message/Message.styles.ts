import styled from "@emotion/styled";
import {
  messageDirectionLeft,
  messageDirectionRight,
} from "../../../styles/utilities/message";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const MessageWrapper = styled.div<{
  direction: string;
  background: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  position: relative;
  padding: 10px;
  text-align: left;
  border-radius: 10px;
  color: white;
  max-width: 200px;
  min-height: 50px;
  background-color: ${(props) => props.background};
  border: 1px solid ${(props) => props.background};

  ${(props: any) => twinStyles(props)};

  &::after,
  &::before {
    content: "";
    position: absolute;
    width: 0;
    height: 0;

    border-left: 16px solid transparent;
    border-right: 16px solid transparent;
  }

  &::before {
    ${(props) =>
      props.direction == "left" && messageDirectionLeft(props, "before")}

    ${(props) =>
      props.direction == "right" && messageDirectionRight(props, "before")}
  }

  &::after {
    ${(props) =>
      props.direction == "left" && messageDirectionLeft(props, "after")}

    ${(props) =>
      props.direction == "right" && messageDirectionRight(props, "after")}
  }
`;
