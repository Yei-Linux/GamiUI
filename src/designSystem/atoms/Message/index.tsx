export { default } from './Message'
export { options } from './constants'