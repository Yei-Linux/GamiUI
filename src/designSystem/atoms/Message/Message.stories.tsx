import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Message from ".";

const Template = getTemplate(Message);

export default {
  title: "Atoms/Message",
  component: Message,
  args: { ...Message.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const BasicMessageLeft = Template.bind({});
BasicMessageLeft.args = {
  width: "NORMAL",
  shadow: "",
};

export const BasicMessageRight= Template.bind({});
BasicMessageRight.args = {
  width: "NORMAL",
  direction: "right",
  shadow: "",
};