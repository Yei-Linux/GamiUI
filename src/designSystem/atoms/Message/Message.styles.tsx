import { css } from '@emotion/react'
import styled from '@emotion/styled'
import {
  FontWeightType,
  HeightType,
  ShadowType,
  TextAlignType,
  WidthType,
} from 'core/domain/types'
import { mixinFlexVariants } from 'styles/mixins/flex'
import { theme } from 'styles/tokens'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'
import { TDirection } from './Message'

const messageDirectionLeft = (
  background: string,
  type: 'before' | 'after'
) => css`
  border-top: ${type == 'before' ? '17px' : '15px'} solid ${background};
  top: ${type == 'before' ? '-1px' : '0'};
  left: ${type == 'before' ? '-17px' : '-15px'};
`

const messageDirectionRight = (
  background: string,
  type: 'before' | 'after'
) => css`
  border-bottom: ${type == 'before' ? '17px' : '15px'} solid ${background};
  bottom: ${type == 'before' ? '-1px' : '0'};
  right: ${type == 'before' ? '-17px' : '-15px'};
`

export const Message = styled.div<{
  $direction: TDirection
  $background: string
  $color?: string
  $maxWidth?: string

  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
  $textAlign?: TextAlignType
  $fontWeight?: FontWeightType
}>`
  position: relative;

  color: ${({ $color }) => $color || theme.light.neutral[800]};
  background-color: ${({ $background }) =>
    $background || theme.light.primary.jordyBlue};
  border: 1px solid
    ${({ $background }) => $background || theme.light.primary.jordyBlue};

  max-width: ${({ $maxWidth }) => $maxWidth};
  min-height: 20px;

  ${mixinFlexVariants({ alignItems: 'center' })}

  &.marker {
    &::after,
    &::before {
      content: '';
      position: absolute;
      width: 0;
      height: 0;

      border-left: 16px solid transparent;
      border-right: 16px solid transparent;
    }
  }

  &.directionLeft {
    &::before {
      ${({ $background }) => messageDirectionLeft($background, 'before')};
    }
    &::after {
      ${({ $background }) => messageDirectionLeft($background, 'after')};
    }
  }

  &.directionRight {
    &::before {
      ${({ $background }) => messageDirectionRight($background, 'before')};
    }
    &::after {
      ${({ $background }) => messageDirectionRight($background, 'after')};
    }
  }

  ${({ $shadow, $fontWeight, $width, $textAlign, $heigth }) =>
    setGenericPropStyles({
      border: 'sm',
      shadow: $shadow || 'md',
      width: $width || 'xs',
      heigth: $heigth || 'xs',
      fontWeight: $fontWeight,
      textAlign: $textAlign || 'left',
    })};
`
