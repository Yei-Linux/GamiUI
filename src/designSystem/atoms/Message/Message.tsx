import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { MessageWrapper } from "./Message.styles";

export interface MessageProps extends IGeneralProps {
  /**
   * Message Text
   */
  text: string;
  /**
   * Message Direction
   */
  direction: "left" | "right";
  /**
   * Background
   */
  background: string;
}

const Message = ({ text, direction, background, ...args }: MessageProps) => (
  <MessageWrapper background={background} direction={direction} {...args}>
    {text}
  </MessageWrapper>
);

Message.defaultProps = {
  text: "Hola Bienvenido a nuestra app!",
  direction: "left",
  background: "#7868e6",
};

export default Message;
