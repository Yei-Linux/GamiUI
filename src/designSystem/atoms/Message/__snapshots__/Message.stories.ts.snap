// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Message Basic Message Left 1`] = `
<div
  className="css-1tfozn8-MessageWrapper e6z3dsf0"
  direction="left"
  width="NORMAL"
>
  Hola Bienvenido a nuestra app!
</div>
`;

exports[`Storyshots Atoms/Message Basic Message Right 1`] = `
<div
  className="css-yqmcn-MessageWrapper e6z3dsf0"
  direction="right"
  width="NORMAL"
>
  Hola Bienvenido a nuestra app!
</div>
`;
